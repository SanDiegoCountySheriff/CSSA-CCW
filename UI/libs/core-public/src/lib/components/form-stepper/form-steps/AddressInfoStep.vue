<template>
  <div>
    <v-form
      ref="form"
      v-model="valid"
    >
      <v-row
        v-if="isRenew"
        justify="center"
        align="center"
      >
        <v-col
          cols="12"
          md="6"
        >
          <v-alert
            type="info"
            color="primary"
            dark
            outlined
            elevation="2"
          >
            Please review your address information and ensure everything is up
            to date before proceeding
          </v-alert>
        </v-col>
      </v-row>
      <v-card-title v-if="!isMobile">
        {{ $t('Address Information') }}
      </v-card-title>

      <v-card-subtitle v-if="isMobile">
        {{ $t('Address Information') }}
      </v-card-subtitle>

      <v-card-text>
        <v-row>
          <v-col
            md="6"
            cols="12"
            :class="isMobile ? 'pb-0' : ''"
          >
            <v-text-field
              v-model="model.application.currentAddress.streetAddress"
              :rules="[v => !!v || $t('Street address cannot be blank')]"
              :label="$t('Street Address')"
              :dense="isMobile"
              maxlength="150"
              outlined
            >
            </v-text-field>
          </v-col>

          <v-col
            md="6"
            cols="12"
            :class="isMobile ? 'pb-0' : ''"
          >
            <v-text-field
              v-model="model.application.currentAddress.city"
              :rules="[v => !!v || $t('City cannot be blank')]"
              :label="$t('City')"
              :dense="isMobile"
              maxlength="100"
              outlined
            >
            </v-text-field>
          </v-col>
        </v-row>

        <v-row>
          <v-col
            md="6"
            cols="12"
            :class="isMobile ? 'pb-0' : ''"
          >
            <v-text-field
              :label="$t('Country')"
              :dense="isMobile"
              :value="
                (model.application.currentAddress.country = 'United States')
              "
              readonly
              :hint="$t('Applicant must reside in the United States')"
              persistent-hint
              outlined
            >
            </v-text-field>
          </v-col>
          <v-col
            md="6"
            cols="12"
            :class="isMobile ? 'pb-0' : ''"
          >
            <v-autocomplete
              v-if="
                model.application.currentAddress.country === 'United States'
              "
              v-model="model.application.currentAddress.state"
              :rules="[v => !!v || $t('State cannot be blank')]"
              :label="$t('State')"
              :dense="isMobile"
              :items="states"
              auto-select-first
              maxlength="100"
              outlined
            >
            </v-autocomplete>

            <v-text-field
              v-if="
                model.application.currentAddress.country !== 'United States'
              "
              v-model="model.application.currentAddress.state"
              :rules="[v => !!v || $t('Region cannot be blank')]"
              :label="$t('Region')"
              :dense="isMobile"
              maxlength="100"
              outlined
            >
            </v-text-field>
          </v-col>
        </v-row>

        <v-row>
          <v-col
            md="6"
            cols="12"
            :class="isMobile ? 'pb-0' : ''"
          >
            <v-text-field
              v-model="model.application.currentAddress.county"
              :rules="[v => !!v || $t('County cannot be blank')]"
              :hint="$t('If not applicable enter N/A ')"
              :label="$t('County')"
              :dense="isMobile"
              persistent-hint
              maxlength="100"
              outlined
            >
            </v-text-field>
          </v-col>
          <v-col
            cols="12"
            md="6"
          >
            <v-text-field
              v-model="model.application.currentAddress.zip"
              :hint="$t('If not applicable enter N/A ')"
              :rules="zipRuleSet"
              :dense="isMobile"
              :label="$t('Zip')"
              persistent-hint
              maxlength="10"
              outlined
            >
            </v-text-field>
          </v-col>
        </v-row>

        <v-checkbox
          v-model="model.application.differentMailing"
          :label="$t('Different Mailing address')"
          @change="handleValidateForm"
        />
      </v-card-text>

      <div v-if="model.application.differentMailing">
        <v-card-title v-if="!isMobile">
          {{ $t('Mailing Address') }}
        </v-card-title>

        <v-card-subtitle v-else>
          {{ $t('Mailing Address') }}
        </v-card-subtitle>

        <v-card-text>
          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-text-field
                v-model="model.application.mailingAddress.streetAddress"
                :rules="[v => !!v || $t('Street address cannot be blank')]"
                :label="$t('Street Address')"
                :dense="isMobile"
                maxlength="150"
                outlined
              >
              </v-text-field>
            </v-col>

            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-text-field
                v-model="model.application.mailingAddress.city"
                :rules="[v => !!v || $t('City cannot be blank')]"
                :label="$t('City')"
                :dense="isMobile"
                maxlength="100"
                outlined
              >
              </v-text-field>
            </v-col>
          </v-row>

          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-combobox
                v-model="model.application.mailingAddress.country"
                :rules="[v => !!v || $t('Country cannot be blank')]"
                :label="$t('Country')"
                :items="countries"
                :dense="isMobile"
                auto-select-first
                outlined
              >
              </v-combobox>
            </v-col>
            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-autocomplete
                v-if="
                  model.application.mailingAddress.country === 'United States'
                "
                v-model="model.application.mailingAddress.state"
                :rules="[v => !!v || $t('State cannot be blank')]"
                :label="$t('State')"
                :dense="isMobile"
                :items="states"
                auto-select-first
                maxlength="100"
                outlined
              >
              </v-autocomplete>

              <v-text-field
                v-if="
                  model.application.mailingAddress.country !== 'United States'
                "
                v-model="model.application.mailingAddress.state"
                :rules="[v => !!v || $t('Region cannot be blank')]"
                :label="$t('Region')"
                :dense="isMobile"
                maxlength="100"
                outlined
              >
              </v-text-field>
            </v-col>
          </v-row>

          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-text-field
                v-model="model.application.mailingAddress.county"
                :rules="[v => !!v || $t('County cannot be blank')]"
                :hint="$t('If not applicable enter N/A ')"
                :label="$t('County')"
                :dense="isMobile"
                persistent-hint
                maxlength="100"
                outlined
              >
              </v-text-field>
            </v-col>
            <v-col
              cols="12"
              md="6"
            >
              <v-text-field
                v-model="model.application.mailingAddress.zip"
                :hint="$t('If not applicable enter N/A')"
                :rules="zipRuleSet"
                :label="$t('Zip')"
                :dense="isMobile"
                persistent-hint
                maxlength="10"
                outlined
              >
              </v-text-field>
            </v-col>
          </v-row>
        </v-card-text>
      </div>

      <v-card-text>
        <v-checkbox
          v-model="model.application.differentSpouseAddress"
          :label="$t('Different Spouse address')"
          @change="handleValidateForm"
        />
      </v-card-text>

      <div v-if="model.application.differentSpouseAddress">
        <v-card-title v-if="!isMobile">
          {{ $t('Spouse Address') }}
        </v-card-title>

        <v-card-subtitle v-else>
          {{ $t('Spouse Address') }}
        </v-card-subtitle>

        <v-card-text>
          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-text-field
                v-model="
                  model.application.spouseAddressInformation.streetAddress
                "
                :rules="[
                  v => !!v || $t('Spouse street address cannot be blank'),
                ]"
                :label="$t('Spouse street address')"
                :dense="isMobile"
                maxlength="150"
                outlined
              ></v-text-field>
            </v-col>

            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-text-field
                v-model="model.application.spouseAddressInformation.city"
                :rules="[v => !!v || $t('Spouse\'s City cannot be blank')]"
                :label="$t('Spouse\'s City')"
                :dense="isMobile"
                maxlength="100"
                outlined
              >
              </v-text-field>
            </v-col>
          </v-row>

          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-combobox
                v-model="model.application.spouseAddressInformation.country"
                :rules="[v => !!v || $t('Spouse\'s Country cannot be blank')]"
                :label="$t('Spouse\'s Country')"
                :items="countries"
                :dense="isMobile"
                auto-select-first
                outlined
              >
              </v-combobox>
            </v-col>
            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-autocomplete
                v-if="
                  model.application.spouseAddressInformation.country ===
                  'United States'
                "
                v-model="model.application.spouseAddressInformation.state"
                :rules="[v => !!v || $t('State cannot be blank')]"
                :label="$t('State')"
                :dense="isMobile"
                :items="states"
                auto-select-first
                maxlength="100"
                outlined
              >
              </v-autocomplete>

              <v-text-field
                v-if="
                  model.application.spouseAddressInformation.country !==
                  'United States'
                "
                v-model="model.application.spouseAddressInformation.state"
                :rules="[v => !!v || $t('Region cannot be blank')]"
                :label="$t('Region')"
                :dense="isMobile"
                maxlength="100"
                outlined
              >
              </v-text-field>
            </v-col>
          </v-row>

          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="isMobile ? 'pb-0' : ''"
            >
              <v-text-field
                v-model="model.application.spouseAddressInformation.county"
                :rules="[v => !!v || $t('Spouse\'s County cannot be blank')]"
                :hint="$t('If not applicable enter N/A')"
                :label="$t('Spouse\'s County')"
                :dense="isMobile"
                persistent-hint
                maxlength="100"
                outlined
              >
              </v-text-field>
            </v-col>
            <v-col
              cols="12"
              md="6"
            >
              <v-text-field
                v-model="model.application.spouseAddressInformation.zip"
                :hint="$t('If not applicable enter N/A')"
                :label="$t('Spouse\'s Zip')"
                :rules="zipRuleSet"
                :dense="isMobile"
                persistent-hint
                maxlength="10"
                outlined
              >
              </v-text-field>
            </v-col>
          </v-row>

          <v-row>
            <v-col
              cols="12"
              md="6"
            >
              <v-text-field
                v-model="model.application.spouseAddressInformation.reason"
                :label="$t('Reason for different spouse address')"
                :rules="requireReasonRuleSet"
                :dense="isMobile"
                persistent-hint
                maxlength="50"
                outlined
              >
              </v-text-field>
            </v-col>
          </v-row>
        </v-card-text>
      </div>
      <v-card-title>
        {{ $t(' Previous Addresses') }}
      </v-card-title>

      <v-card-text>
        <v-radio-group
          v-model="hasPreviousAddresses"
          :label="'Have you lived anywhere else besides your current residence in the last 5 years?'"
          :row="!isMobile"
          :rules="previousAddressRules"
        >
          <v-radio
            color="primary"
            :label="$t('Yes')"
            :value="true"
          ></v-radio>
          <v-radio
            color="primary"
            :label="$t('No')"
            :value="false"
          ></v-radio>
        </v-radio-group>
      </v-card-text>

      <v-card-text v-if="hasPreviousAddresses">
        <p>
          {{ $t('Please provide residences for the ') }}
          <strong>{{ $t('past 5 years') }}</strong>
        </p>

        <PreviousAddressDialog
          @get-previous-address-from-dialog="getPreviousAddressFromDialog"
        />
        <AddressTable
          :addresses="model.application.previousAddresses"
          :enable-delete="true"
          @delete="deleteAddress"
        />
      </v-card-text>

      <FormButtonContainer
        :valid="valid"
        @continue="handleContinue"
        @save="handleSave"
      />
    </v-form>
  </div>
</template>

<script setup lang="ts">
import AddressTable from '@shared-ui/components/tables/AddressTable.vue'
import { CompleteApplication } from '@shared-utils/types/defaultTypes'
import FormButtonContainer from '@shared-ui/components/containers/FormButtonContainer.vue'
import PreviousAddressDialog from '@shared-ui/components/dialogs/PreviousAddressDialog.vue'
import { requireReasonRuleSet } from '@shared-ui/rule-sets/ruleSets'
import { useVuetify } from '@shared-ui/composables/useVuetify'
import { zipRuleSet } from '@shared-ui/rule-sets/ruleSets'
import {
  AddressInfoType,
  ApplicationType,
} from '@shared-utils/types/defaultTypes'
import { computed, nextTick, onMounted, ref, watch } from 'vue'
import { countries, states } from '@shared-utils/lists/defaultConstants'

interface FormStepThreeProps {
  value: CompleteApplication
}

const props = defineProps<FormStepThreeProps>()
const emit = defineEmits([
  'input',
  'handle-continue',
  'handle-save',
  'update-step-three-valid',
])

const form = ref()
const valid = ref(false)
const hasPreviousAddresses = ref(false)
const vuetify = useVuetify()
const isMobile = computed(
  () => vuetify?.breakpoint.name === 'sm' || vuetify?.breakpoint.name === 'xs'
)

const model = computed({
  get: () => props.value,
  set: (value: CompleteApplication) => emit('input', value),
})

const isRenew = computed(() => {
  const applicationType = model.value.application.applicationType

  return (
    applicationType === ApplicationType['Renew Standard'] ||
    applicationType === ApplicationType['Renew Reserve'] ||
    applicationType === ApplicationType['Renew Judicial'] ||
    applicationType === ApplicationType['Renew Employment']
  )
})

const previousAddressRules = computed(() => {
  if (
    hasPreviousAddresses.value === true &&
    model.value.application.previousAddresses.length === 0
  ) {
    return ['Previous address is required']
  }

  return []
})

onMounted(() => {
  if (form.value) {
    form.value.validate()
  }

  if (model.value.application.previousAddresses.length > 0) {
    hasPreviousAddresses.value = true
  }
})

function getPreviousAddressFromDialog(address: AddressInfoType) {
  model.value.application.previousAddresses.push(address)
}

function deleteAddress(index) {
  model.value.application.previousAddresses.splice(index, 1)
}

function handleContinue() {
  emit('handle-continue')
}

function handleSave() {
  emit('handle-save')
}

function handleValidateForm() {
  if (form.value) {
    nextTick(() => {
      form.value.validate()
    })
  }
}

watch(valid, (newValue, oldValue) => {
  if (newValue !== oldValue) {
    emit('update-step-three-valid', newValue)
  }
})
</script>
