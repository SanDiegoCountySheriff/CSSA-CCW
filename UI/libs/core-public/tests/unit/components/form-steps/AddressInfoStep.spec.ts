// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import AddressInfoStep from '@core-public/components/form-stepper/form-steps/AddressInfoStep.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

const pinia = createTestingPinia();
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('AddressInfoStep', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = mount(AddressInfoStep, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
      propsData: {
        handleNextSection: () => null,
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should render all the inital text inputs', () => {
    expect(wrapper.findAllComponents('.v-text-field')).toHaveLength(7);
  });

  it('Should render the next text inputs', async () => {
    await wrapper.find('#different-mailing').setChecked();
    expect(wrapper.findAllComponents('.v-text-field')).toHaveLength(14);
  });
});
