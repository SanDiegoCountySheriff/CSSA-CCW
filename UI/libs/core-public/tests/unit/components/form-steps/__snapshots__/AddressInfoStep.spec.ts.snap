// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressInfoStep Should match the snapshot 1`] = `
<div>
  <div class="v-subheader sub-header font-weight-bold theme--light">
    Address Information
  </div>
  <form novalidate="novalidate" class="v-form">
    <div class="row ml-5">
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-4" class="v-label theme--light" style="left: 0px; position: absolute;">Address line 1</label><input maxlength="150" id="input-4" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 150</div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-8" class="v-label theme--light" style="left: 0px; position: absolute;">Address line 2</label><input maxlength="150" id="input-8" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 150</div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row ml-5">
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-11" class="v-label theme--light" style="left: 0px; position: absolute;">City</label><input maxlength="100" id="input-11" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 100</div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-15" class="v-label theme--light" style="left: 0px; position: absolute;">State</label><input maxlength="100" id="input-15" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 100</div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-19" class="v-label theme--light" style="left: 0px; position: absolute;">County</label><input maxlength="100" id="input-19" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 100</div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-23" class="v-label theme--light" style="left: 0px; position: absolute;">Zip</label><input maxlength="10" id="input-23" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 10</div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-27" class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-select__slot"><label for="input-27" class="v-label theme--light" style="left: 0px; position: absolute;">Country</label><input id="input-27" type="text" autocomplete="off">
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="row ml-5 my-5">
      <div class="col-lg-6 col-12">
        <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="different-mailing" role="checkbox" type="checkbox" value="">
                <div class="v-input--selection-controls__ripple"></div>
              </div><label for="different-mailing" class="v-label theme--light" style="left: 0px; position: relative;">Different Mailing address</label>
            </div>
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="different-spouse" role="checkbox" type="checkbox" value="">
                <div class="v-input--selection-controls__ripple"></div>
              </div><label for="different-spouse" class="v-label theme--light" style="left: 0px; position: relative;">Different Spouse address</label>
            </div>
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
    <!---->
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Previous Address
    </div>
    <p>Please provide residenses for the past 5 years</p>
    <div class="previous-address-container">
      <div class="container container--fluid">
        <div class="v-data-table my-3 theme--light">
          <div class="v-data-table__wrapper">
            <table>
              <thead>
                <tr>
                  <th scope="col"> Address line 1</th>
                  <th scope="col">Address line 2</th>
                  <th scope="col">City</th>
                  <th scope="col">State</th>
                  <th scope="col">County</th>
                  <th scope="col">Zip</th>
                  <th scope="col">Country</th>
                  <th scope="col"></th>
                </tr>
              </thead>
              <tbody></tbody>
            </table>
          </div>
        </div>
      </div>
      <div><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary" id="add-previous-address-btn" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content">
         Add previous address
      </span></button>
        <div class="v-dialog__container">
          <!---->
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="my-5 v-divider theme--light">
    <div class="form-btn-container">
      <div class="form-btn-inner">
        <div><button type="button" disabled="disabled" class="mt-3 v-btn v-btn--disabled v-btn--has-bg theme--light v-size--small"><span class="v-btn__content">
        Continue
      </span></button> <button type="button" disabled="disabled" class="mt-3 v-btn v-btn--disabled v-btn--has-bg theme--light v-size--small"><span class="v-btn__content">
        Save and Exit
      </span></button></div>
        <div><button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small warning mx-2"><span class="v-btn__content">
        Go back
      </span></button> <button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small error mr-2"><span class="v-btn__content">
        Cancel
      </span></button></div>
      </div>
    </div>
  </form>
  <div class="v-snack v-snack--bottom">
    <transition-stub name="v-snack-transition">
      <div class="v-snack__wrapper v-sheet v-sheet--outlined theme--light error--text" style="display: none;">
        <div role="status" aria-live="polite" class="v-snack__content">
          Section update unsuccessful please try again.
        </div>
        <div class="v-snack__action "></div>
      </div>
    </transition-stub>
  </div>
</div>
`;
